packages/compose/src/hooks/use-copy-to-clipboard/index.js

```diff
@@ -39,32 +34,34 @@ export default function useCopyToClipboard( text, onSuccess ) {
 	// fresh when the callback is called.
 	const textRef = useUpdatedRef( text );
 	const onSuccessRef = useUpdatedRef( onSuccess );
-	return useRefEffect( ( node ) => {
-		// Clipboard listens to click events.
-		const clipboard = new Clipboard( node, {
-			text() {
-				return typeof textRef.current === 'function'
+	return useRefEffect( ( trigger ) => {
+		if ( ! trigger ) {
+			return;
+		}
+
+		const copyTextToClipboard = () => {
+			const currentWindow = trigger.ownerDocument.defaultView || window;
+			const textToCopy =
+				typeof textRef.current === 'function'
 					? textRef.current()
 					: textRef.current || '';
-			},
-		} );
 
-		clipboard.on( 'success', ( { clearSelection } ) => {
-			// Clearing selection will move focus back to the triggering
-			// button, ensuring that it is not reset to the body, and
-			// further that it is kept within the rendered node.
-			clearSelection();
-			// Handle ClipboardJS focus bug, see
-			// https://github.com/zenorocha/clipboard.js/issues/680
-			node.focus();
+			currentWindow?.navigator?.clipboard
+				?.writeText( textToCopy )
+				.then( () => {
+					// Clear selection and ensure focus stays on the trigger element.
+					currentWindow?.getSelection()?.removeAllRanges();
+					trigger.focus();
```

Comment:
Same question as in [this comment](https://github.com/WordPress/gutenberg/pull/37713/files?diff=split&w=0#r778404668)